client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/async/DefaultAsyncExecutorFactory.java

```diff
@@ -58,4 +58,8 @@ static final class ISPNNonBlockingThreadGroup extends ThreadGroup implements Non
          super(name);
       }
    }
+
+   static class NonBlockingThreadGroupHolder {
```

Comment:
We don't need this one, just use the one at DefaultThreadFactory#NonBlockingThreadGroupHolder and make it public.

Reply:
This one doesn't have visibility of DefaultThreadFactory - this is the hotrod-client module, while DefaultThreadFactory is the core module.

Reply:
Ah, yes, definitely. I am thinking it might be better to have a ThreadGroups class in the commons package that holds these so both core and client can reuse the same thread groups. This way if anyone is using a remote store it will share the thread groups.

Reply:
Sounds good, I will look at that.

Reply:
It's updated, but it's pretty late here so I better read it again in the morning.